### ISSUE TITLE
Keyboard shortcuts for Find & Replace In-View Options

### ISSUE THREAD

**Issue by prasannavl** [Microsoft/vscode#406]
Options inside the Find & Replace View have no keyboard shortcuts. There's no way to switch selection, casing, regex with the keyboard, and have to manually click on them using the mouse. 

Its one of the only place I find myself being forced to actually use the mouse. And I believe I'm correct in saying its a bug, since VSCode is a keyboard centric editor. 

PS: Also seems like the design of this box just got directly inherited from VS. A larger box like Atom, Sublime style, or with support for multi-line would more comfortable to work with.

**Comment by rhbecker** [comment]
The more keybinding options the better, imo, but I'm primarily +1'ing because I'd particularly like to see one for executing a replacement.

It seems like re-binding `ctrl+h` would make sense, but with a more specific `when` context, like `"when": "editorTextFocus && editorHasSelection"`. And ideally, there would be a backward advancing alternate, bound to, say, `ctrl+shift+h`.

**Comment by alexandrudima** [comment]
@prasannavl 
I've pushed with 0b90eb2c25202d6379ec180d0d29514c2ee607ef:
* `toggleFindCaseSensitive`: <kbd>ALT</kbd>+<kbd>C</kbd>, mac: <kbd>CMD</kbd>+<kbd>ALT</kbd>+<kbd>C</kbd>
* `toggleFindWholeWord`: <kbd>ALT</kbd>+<kbd>W</kbd>, mac: <kbd>CMD</kbd>+<kbd>ALT</kbd>+<kbd>W</kbd>
* `toggleFindRegex`: <kbd>ALT</kbd>+<kbd>R</kbd>, mac: <kbd>CMD</kbd>+<kbd>ALT</kbd>+<kbd>R</kbd>

These will also work when the find widget is not visible.

@rhbecker The replace can be triggered by <kbd>Enter</kbd> when focus is in the replace field. I've also added that <kbd>CTRL</kbd>+<kbd>ArrowDown</kbd> focuses the editor from the find widget. I wasn't sure from your comment, would you want something more?

**Comment by prasannavl** [comment]
@alexandrudima - Looks good! Thanks for quickly resolving this. I think that should do it for the most part.

From a quick glance at the commit, I'm not sure if the `when` context as suggested by @rhbecker has been added, but I think that's quite important to ensure that the editor shortcuts aren't polluted when the find box is not open.

**Comment by rhbecker** [comment]
@alexandrudima 

Sorry, my last comment seems a bit off, in retrospect.

What I should have suggested is re-binding <kbd>Ctrl</kbd>+<kbd>h</kbd> and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>h</kbd>, with context ...
`"when": "editorFocus && findWidgetVisible"`.

Both would execute the replacement already entered into the find / replace fields, regardless of which of the fields in the find widget has focus. The former would replace and advance to the next match. The latter would replace and advance to the prior match.

Suppose you ...

1. <kbd>Ctrl</kbd>+<kbd>h</kbd> to open the find widget.
2. Input string to find, or accept string already present from prior find, or because it was highlighted when you entered the widget.
3. <kbd>Tab</kbd> to reach the replace field.
4. Input replacement text.

Presently, you can only execute the replacement if you stay in that replace field, from where you can hit <kbd>Enter</kbd>. But what if you <kbd>Tab</kbd> back to the search field - maybe to update the search string? Both fields are populated. Seems like it should be possible to execute without first switching back to the replace field.

Probably more importantly, suppose you <kbd>Esc</kbd> out of the find widget entirely. The first press of <kbd>Ctrl</kbd>+<kbd>h</kbd> re-opens the widget, with previous values still populating the find / replace fields. It would be nice if you could then immediately re-press <kbd>Ctrl</kbd>+<kbd>h</kbd> to execute, without having to <kbd>Tab</kbd>, then <kbd>Enter</kbd>.

And of course, switching replacement execution to a keybinding would then allow for a binding to advance backward after each replacement.

Hopefully that all makes sense?

**Comment by rhbecker** [comment]
The changes provided via the December release are probably sufficient, with respect to my suggestions.

The only thing that remains odd to me - and, based on his earlier comment, apparently @prasannavl as well - is that the <kbd>alt</kbd>+<kbd>c</kbd>, <kbd>alt</kbd>+<kbd>r</kbd>, and <kbd>alt</kbd>+<kbd>w</kbd> additions aren't scoped to a more specific context, such as `"when": "editorFocus && findWidgetVisible"`.

But, if there's some good reason to leave it more open, it's probably not worth addressing.

In other words, don't leave this one open on my behalf.

**Comment by alexandrudima** [comment]
@rhbecker Thank you for the feedback and I'm happy you're happy!

I've left <kbd>alt</kbd>+<kbd>c</kbd>, <kbd>alt</kbd>+<kbd>r</kbd> and <kbd>alt</kbd>+<kbd>w</kbd> intentionally without the `findWidgetVisible` context condition because the find widget modifiers also affect <kbd>Ctrl</kbd>+<kbd>D</kbd> and <kbd>Ctrl</kbd>+<kbd>K</kbd> <kbd>Ctrl</kbd>+<kbd>D</kbd>.

To see what I mean, check out the following snippet:
```cpp
#include <iostream>

int main() {
    cout << "hello world"; cout << endl;
    cout << "hello world again" << endl;
    cout << "Hello world again" << endl;
    cout << "Helloworld again" << endl;
}
```

Without opening the find widget whatsoever, double click on the first `hello`. All 4 other instances of `hello` (regardless of casing) should get a highlight and <kbd>Ctrl</kbd>+<kbd>D</kbd> will reach them. Now select again the first `hello` and press <kbd>alt</kbd>+<kbd>c</kbd>. Only the first two instances of `hello` (same case) should get a highlight and be reachable via <kbd>Ctrl</kbd>+<kbd>D</kbd>. If you reveal the find widget, you will notice the case modifier is now toggled on.

**Comment by rhbecker** [comment]
@alexandrudima Ah, that makes sense and is very handy. Thanks for the tip!
